### ISSUE TITLE
About Dataset's reproducibility.

### ISSUE THREAD

**Issue by FFFFlint** [tsinghua-fib-lab/DGCN#12]
Thank you for the nice work! Now I'm working on a project that needs other features from 'UserBehavior.csv' (e.g., the timestamp ), and I tried to generate the processed dataset by using the service [https://github.com/DavyMorgan/dps](url), however, for the same file `UserBehavior.csv`, the size of my processed train/val/test set are all different from yours. So I'm wondering whether there is reproducibility of the data processing (maybe it could be the random seed of `dataframe.sample()` when performing downsampling of user)? I would be very grateful if you could provide such reproducibility of dataset!

**Comment by DavyMorgan** [comment]
Hi, here is the snippet that processes the UserBehavior.csv which you can refer to:
```python
def filter_items_with_multiple_cids_taobao_ctr(flags_obj, record):

    item_cate = record[['iid', 'cid']].drop_duplicates().groupby('iid').count().reset_index().rename(columns={'cid': 'count'})
    items_with_single_cid = item_cate[item_cate['count'] == 1]['iid']

    id_filter = utils.FILTER.IDFilter(flags_obj, record)
    record = id_filter.filter(record, 'iid', items_with_single_cid)

    return record


def process_taobao(flags_obj):

    record = utils.load_csv(flags_obj, 'UserBehavior.csv', header=None, names=['uid', 'iid', 'cid', 'behavior', 'ts'])
    record = utils.filter_duplication(flags_obj, record)
    record = filter_items_with_multiple_cids_taobao_ctr(flags_obj, record)
    record = utils.downsample_user(flags_obj, record, 0.1)
    record = utils.filter_cf(flags_obj, record, 10)
    record, user_reindex_map, item_reindex_map = utils.reindex_user_item(flags_obj, record)
    record, cate_reindex_map = utils.reindex_feature(flags_obj, record, 'cid')
    utils.save_reindex_user_item_map(flags_obj, user_reindex_map, item_reindex_map)
    utils.save_reindex_feature_map(flags_obj, 'cate', cate_reindex_map)
    train_record, val_record, test_record = utils.split(flags_obj, record, [0.6, 0.2, 0.2])
    utils.save_csv_record(flags_obj, record, train_record, val_record, test_record)
    utils.report(flags_obj, record)
    utils.extract_save_item_feature(flags_obj, record, 'cate', 'cid')
    coo_record, train_coo_record, val_coo_record, test_coo_record = utils.generate_coo(flags_obj, record, train_record, val_record, test_record)
    utils.save_coo(flags_obj, coo_record, train_coo_record, val_coo_record, test_coo_record)
```

Please do check the codes as the API may be a little bit different from dps.

**Comment by FFFFlint** [comment]
Thank you for your reply! I used this part of the code and found that the problem is the ratio of down-sampled users. In [https://github.com/tsinghua-fib-lab/DGCN/blob/main/data/taobao.py](url), the ratio is 0.05(`record = utils.downsample_user(flags_obj, record, 0.05)`), and after I changed  0.05 into 0.1(`record = utils.downsample_user(flags_obj, record, 0.1)`), I've got the data with the right size, thank you for your generous help again!
